JUnit TestSuite not handled correctly in Reports

### TestNG Version
> Note: only the latest version is supported
Broken with 6.9.13.3 (6.9.10 is latest version where this works)

### Expected behavior
JUnit TestSuite classes should be run, and failed test methods should be reported as failures. Even if one uses JUnit 4+.

### Actual behavior
JUnit TestCase classes are run and results are reported correctly whereas JUnit TestSuite classes are run but failed test methods and successful methods are not reported.

(!) Note that this only happens with JUnit 4+ on the classpath, not with JUnit 3!

There are Exceptions in reporters:

1. jq Reporter
[TestNG] Reporter org.testng.reporters.jq.Main@1dd02175 failed
java.lang.NullPointerException
        at java.base/java.lang.StringBuilder.<init>(StringBuilder.java:124)
        at org.testng.reporters.jq.Model.getTestResultName(Model.java:142)
        at org.testng.reporters.jq.ChronologicalPanel.getContent(ChronologicalPanel.java:68)
        at org.testng.reporters.jq.BaseMultiSuitePanel.generate(BaseMultiSuitePanel.java:24)
        at org.testng.reporters.jq.Main.generateReport(Main.java:63)
        at org.testng.TestNG.generateReports(TestNG.java:1175)
        at org.testng.TestNG.run(TestNG.java:1102)
        at org.testng.TestNG.privateMain(TestNG.java:1425)
        at org.testng.TestNG.main(TestNG.java:1394)

2. SuiteHTMLReporter
[TestNG] Reporter org.testng.reporters.SuiteHTMLReporter@4df828d7 failed
java.lang.NullPointerException
        at org.testng.reporters.SuiteHTMLReporter$1.compare(SuiteHTMLReporter.java:366)
        at java.base/java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
        at java.base/java.util.TimSort.sort(TimSort.java:220)
        at java.base/java.util.Arrays.sort(Arrays.java:1515)
        at java.base/java.util.ArrayList.sort(ArrayList.java:1749)
        at java.base/java.util.Collections$SynchronizedList.sort(Collections.java:2468)
        at java.base/java.util.Collections.sort(Collections.java:177)
        at org.testng.reporters.SuiteHTMLReporter.generateMethodsChronologically(SuiteHTMLReporter.java:369)
        at org.testng.reporters.SuiteHTMLReporter.generateReport(SuiteHTMLReporter.java:73)
        at org.testng.TestNG.generateReports(TestNG.java:1175)
        at org.testng.TestNG.run(TestNG.java:1102)
        at org.testng.TestNG.privateMain(TestNG.java:1425)
        at org.testng.TestNG.main(TestNG.java:1394)


### Is the issue reproductible on runner?
I just tested with eclipse (testng plugin 6.14.3) and Testng.main "path/to/suite.xml"

- [x] Shell
- [ ] Maven
- [x] Gradle
- [ ] Ant
- [x] Eclipse
- [ ] IntelliJ
- [ ] NetBeans


### Test case sample
> Please, share the test case (as small as possible) which shows the issue

Instructions:
Extract zip and run with
>    gradlew rii

To see how a correct report should look like change the testng version in test/build.gradle to e.g. 6.9.10

Note that the test suite contains a test method that should fail and therefore we expect the task to fail if testng works correctly!

[minimal.zip](https://github.com/cbeust/testng/files/3136964/minimal.zip)

The latest version seems to have changed w.r.t. org.testng.annotations.Test, this annotation is missing in the jar file!

I tried with repositories:
1. jcenter()
2. mavencentral()

Both contain an 7.0.0-beta4 artifact without the annotation! Was this annotation intentionally removed or are these artifacts broken? E.g. the jar at http://jcenter.bintray.com/org/testng/testng/7.0.0-beta4/

When using
maven {  url 'https://oss.sonatype.org/content/repositories/snapshots' }

with 7.0.0-SNAPSHOT (testng-7.0.0-20190501.063748-94.jar) I see the org.testng.annotations.Test annotation but the issue still occurs!

With 7.0.0-beta5 it also still occurs.

With 7.0.0-beta6 I'm getting a strange error:
`Error: Unable to initialize main class org.testng.TestNG
Caused by: java.lang.NoClassDefFoundError: com/beust/jcommander/ParameterException`

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -1,4 +1,5 @@
 Current
+Fixed: GITHUB-2069: JUnit TestSuite not handled correctly in Reports (Krishnan Mahadevan)
 Fixed: GITHUB-2075: Thread interrupt flag persists between test methods (Krishnan Mahadevan)
 Fixed: GITHUB-2074: Thread Interrupted when using expectedException (Krishnan Mahadevan)
 Fixed: GITHUB-2802: some methods' javadoc in ISuiteListener and ITestListener is misleading (Yehui Wang)

--- a/src/main/java/org/testng/reporters/SuiteHTMLReporter.java
+++ b/src/main/java/org/testng/reporters/SuiteHTMLReporter.java
@@ -332,6 +332,13 @@ public class SuiteHTMLReporter implements IReporter {
 
   public static final String BEFORE = "&gt;&gt;";
 
+  private static String getMethodName(String name) {
+    if (name == null) {
+      return "";
+    }
+    return name;
+  }
+
   private void generateMethodsChronologically(
       XmlSuite xmlSuite, ISuite suite, String outputFileName, boolean alphabetical) {
     try (BufferedWriter bw = Utils.openWriter(getOutputDirectory(xmlSuite), outputFileName)) {
@@ -348,7 +355,7 @@ public class SuiteHTMLReporter implements IReporter {
 
       List<IInvokedMethod> invokedMethods = suite.getAllInvokedMethods();
       if (alphabetical) {
-        invokedMethods.sort(Comparator.comparing(o -> o.getTestMethod().getMethodName()));
+        invokedMethods.sort(Comparator.comparing(o -> getMethodName(o.getTestMethod().getMethodName())));
       }
 
       SimpleDateFormat format = new SimpleDateFormat("yy/MM/dd HH:mm:ss");

--- a/src/main/java/org/testng/reporters/jq/Model.java
+++ b/src/main/java/org/testng/reporters/jq/Model.java
@@ -133,8 +133,15 @@ public class Model {
     return m_model.get(suite);
   }
 
+  private static String getMethodName(String name) {
+    if (name == null) {
+      return "";
+    }
+    return name;
+  }
+
   public static String getTestResultName(ITestResult tr) {
-    StringBuilder result = new StringBuilder(tr.getMethod().getMethodName());
+    StringBuilder result = new StringBuilder(getMethodName(tr.getMethod().getMethodName()));
     Object[] parameters = tr.getParameters();
     if (parameters.length > 0) {
       result.append("(");

--- a/src/test/java/test/reports/issue2069/Dummy1.java
+++ b/src/test/java/test/reports/issue2069/Dummy1.java
@@ -1,0 +1,15 @@
+package test.reports.issue2069;
+
+import junit.framework.Test;
+import junit.framework.TestSuite;
+
+public class Dummy1 extends TestSuite {
+
+  public static Test suite() {
+    final TestSuite suite = new TestSuite("Failing report");
+    suite.addTestSuite(Dummy3.class);
+    return suite;
+  }
+
+}
+

--- a/src/test/java/test/reports/issue2069/Dummy2.java
+++ b/src/test/java/test/reports/issue2069/Dummy2.java
@@ -1,0 +1,10 @@
+package test.reports.issue2069;
+
+import junit.framework.TestCase;
+
+public class Dummy2 extends TestCase {
+
+  public void testScheduleDelete() {
+  }
+}
+

--- a/src/test/java/test/reports/issue2069/Dummy3.java
+++ b/src/test/java/test/reports/issue2069/Dummy3.java
@@ -1,0 +1,19 @@
+package test.reports.issue2069;
+
+import junit.framework.TestCase;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class Dummy3 extends TestCase {
+
+  @Test
+  public void testFails() {
+    Assert.assertEquals("test3 assertion that will fail", "1", "2");
+  }
+
+  @Test
+  public void testSucceeds() {
+    Assert.assertEquals("test3 assertion", "1", "1");
+  }
+}
+

--- a/src/test/java/test/reports/issue2069/Dummy4.java
+++ b/src/test/java/test/reports/issue2069/Dummy4.java
@@ -1,0 +1,10 @@
+package test.reports.issue2069;
+
+import org.testng.annotations.Test;
+
+public class Dummy4 {
+
+  @Test
+  public void testngTest() {
+  }
+}

--- a/src/test/java/test/reports/issue2069/IssueTest.java
+++ b/src/test/java/test/reports/issue2069/IssueTest.java
@@ -1,0 +1,45 @@
+package test.reports.issue2069;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import static org.assertj.core.api.Assertions.assertThat;
+import org.testng.TestNG;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.xml.XmlSuite;
+import test.SimpleBaseTest;
+
+public class IssueTest extends SimpleBaseTest {
+
+  private PrintStream currentError = System.err;
+  private final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+  @BeforeMethod
+  public void setup() throws UnsupportedEncodingException {
+    PrintStream ps = new PrintStream(baos, true, "UTF-8");
+    System.setErr(ps);
+  }
+
+  @AfterMethod(alwaysRun = true)
+  public void teardown() {
+    System.setErr(currentError);
+  }
+
+  @Test
+  public void ensureNoExceptionsAriseFromReporters() {
+    XmlSuite xmlSuite = createXmlSuite("Not Failing TestSuite");
+    createXmlTest(xmlSuite, "TestngTest", Dummy4.class);
+    createXmlTest(xmlSuite, "TestSuite", Dummy1.class).setJunit(true);
+    createXmlTest(xmlSuite, "TestCase", Dummy2.class).setJunit(true);
+    TestNG tng = create(xmlSuite);
+    tng.setVerbose(2);
+    tng.setUseDefaultListeners(true);
+    tng.run();
+    String data = new String(baos.toByteArray(), StandardCharsets.UTF_8);
+    assertThat(data).doesNotContain("NullPointerException");
+  }
+
+}

--- a/src/test/resources/testng.xml
+++ b/src/test/resources/testng.xml
@@ -112,6 +112,7 @@
       <class name="test.simple.IncludedExcludedTest" />
       <class name="test.reports.ReportTest" />
       <class name="test.reports.XmlReporterTest"/>
+      <class name="test.reports.issue2069.IssueTest"/>
       <class name="test.annotationtransformer.AnnotationTransformerTest" />
 <!--
       <class name="test.jar.JarTest" />

